src/main/java/com/hazelcast/kubernetes/HazelcastKubernetesDiscoveryStrategyFactory.java

```diff
@@ -67,4 +69,24 @@ public DiscoveryStrategy newDiscoveryStrategy(DiscoveryNode discoveryNode, ILogg
     public Collection<PropertyDefinition> getConfigurationProperties() {
         return PROPERTY_DEFINITIONS;
     }
+
+    /**
+     * In all Kubernetes environments the file "/var/run/secrets/kubernetes.io/serviceaccount/token" is injected into the
+     * container. That is why we can use it to verify if this code is run in the Kubernetes environment.
+     * <p>
+     * Note that if the Kubernetes environment is not configured correctly, this file my not exist. However, in such case,
+     * this plugin won't work anyway, so it makes perfect sense to return {@code false}.
+     *
+     * @return true if running in the Kubernetes environment
+     */
+    @Override
+    @SuppressFBWarnings("DMI_HARDCODED_ABSOLUTE_FILENAME")
+    public boolean isAutoDetectionApplicable() {
+        return new File("/var/run/secrets/kubernetes.io/serviceaccount/token").exists();
```

Comment:
Use `Files.exists(path)` instead. Otherwise, you'd need to check if the file is not a directory as well.

Reply:
Isn't it actually slower? https://rules.sonarsource.com/java/tag/performance/RSPEC-3725

FYI: This will never be a directory, so not a big deal with that.

Reply:
As I see there used to be a bug on JDK 8 which got fixed later on. However, I don't see any signs that it was backported JDK 8, so I guess we can leave it as is 🤷‍♂️